explainer.md

```diff
@@ -0,0 +1,208 @@
+# WebXR Device API - Hand Input
+
+This document describes a design giving developers access to hand-tracking XR systems, building on top of the [WebXR device API](https://immersive-web.github.io/webxr/)
+
+## Use cases and scope
+
+This API primarily exposes the poses of hand skeleton joints. It can be used to render a hand model in VR scenarios, as well perform gesture detection with the hands. It does not provide access to a full hand mesh.
+
+## Accessing this API
+
+This API will only be accessible if a `"hand-tracking"` [XR feature](https://immersive-web.github.io/webxr/#feature-dependencies) is requested.
+
+This API presents itself as an additional field on `XRInputSource`, `hand`. The `hand` attribute will be non-null if the input source supports hand tracking and the feature has been requested.
+
+```js
+navigator.xr.requestSession({optionalFeatures: ["hand-tracking"]}).then(...);
+
+function renderFrame(session, frame) {
+   // ...
+
+   for (inputSource of session.inputSources) {
+      if (inputSource.hand) {
+         // render a hand model
+         // perform gesture detection
+      }
+   }
+}
+
+
+```
+
+## Hands and joints
+
+Each hand is made up many bones, connected by _joints_. We name them with their connected bone, for example `INDEX_PHALANGE_DISTAL` is the joint closer to the wrist connected to the distal phalange bone of the index finger. The `*_PHALANGE_TIP` "joints" locate the tips of the fingers. The `WRIST` joint is located at the composite joint between the wrist and forearm.
+
+The joints can be accessed via indexing, for example to access the middle knuckle joint one would use:
+
+```js
+let joint = inputSource.hand[MIDDLE_PHALANGE_PROXIMAL];
+```
+
+Not all devices support all joints, this indexing getter will return `null` when accessing a joint that is not supported by the current user agent or device. This will not change for a given input source. If a joint is supported but not currently being tracked, the getter will still produce the `XRJoint`, with its associated `space` returning `null` when run through `getPose` (etc).
+
+Each joint has an `XRSpace` `space`, located at its center, with its `-Y` direction pointing perpendicular to the skin, outwards from the palm, and `+Z` direction pointing along their associated bone, away from the wrist. This space will return null poses when the joint loses tracking.
```

Comment:
The orientation matches what openxr does, except for the sign of the Z axis.

Reply:
Why is the sign of the Z axis different? That seems like a potential point of confusion and pit of failure, given that it's inverted from every other space in the API.

Reply:
I don't mind changing it. It feels weird to have the Z axis point _away_ from the finger instead of _along_ it. @yl-msft was there a reason for this design decision.

Reply:
In WebXR (and OpenXR):
* +X is right
* +Y is up
* -Z is forward (+Z is backward)

Forward being defined as -Z rather than +Z is simply a consequence of using right-handed coordinate systems.  Agreed that we should not switch to a left-handed coordinate system just for hand joint data.

Reply:
Oh, I was basically suggesting to still use a right handed system, but `+X` points left instead. That's the least natural of these vectors, it's nice to pin the other two to physical quantities and then say "`+X` is the leftover one".

But I'll just change it, this is no big deal.